### PR TITLE
More robust team exists api endpoint

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
More robust team exists api endpoint

#### Ticket Link
[MM-18230](https://mattermost.atlassian.net/browse/MM-18230)

### CODE REVIEW

File: api4/team.go [hunk 1]
```diff
@@ -806,10 +806,17 @@ func teamExists(c *Context, w http.ResponseWriter, r *http.Request) {
...
-	} else {
+	} else if _, err := c.App.GetTeamMember(team.Id, c.App.Session.UserId); err == nil {
+		resp["exists"] = true
+	} else if team.AllowOpenInvite && c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_LIST_PUBLIC_TEAMS) {
 		resp["exists"] = true
+	} else if !team.AllowOpenInvite && c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_LIST_PRIVATE_TEAMS) {
+		resp["exists"] = true
+	} else {
+		resp["exists"] = false
 	}
```
Comment (reflog) [inline api4/team.go]: A small nit - wouldn't it be nicer to do ` exists = true/false ` in all the ifs and assign at the end `resp["exists"] = exists`? Not important, just cleaner I believe. 0/5

Comment (jespino) [reply api4/team.go]: Yes, I agree. I'll do that, thanks!

Comment (jespino) [reply api4/team.go]: Done

**Comment by amyblais** [comment]
@mkraft Kind reminder to review for v5.16.

**Comment by amyblais** [comment]
@jespino This hasn't been cherry-picked yet.

**Comment by jespino** [comment]
@amyblais Yes, the automatic cherry-pick PR creation didn't work, now is cherry-picked in the release-5.16 branch, thanks for the heads up.

**Pull request merged**